src/find/matchers/delete.rs

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of the uutils findutils package.
+ *
+ * (c) Arcterus <arcterus@mail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use std::fs::{self, FileType};
+use std::io::{self, stderr, Write};
+use std::path::Path;
+
+use walkdir::DirEntry;
+
+use super::{Matcher, MatcherIO};
+
+pub struct DeleteMatcher;
+
+impl DeleteMatcher {
+    pub fn new() -> DeleteMatcher {
+        DeleteMatcher
+    }
+
+    pub fn new_box() -> io::Result<Box<dyn Matcher>> {
+        Ok(Box::new(DeleteMatcher::new()))
+    }
+
+    fn delete(&self, file_path: &Path, file_type: FileType) -> io::Result<()> {
+        if file_type.is_dir() {
+            fs::remove_dir(file_path)
+        } else {
+            fs::remove_file(file_path)
+        }
+    }
+}
+
+impl Matcher for DeleteMatcher {
+    fn matches(&self, file_info: &DirEntry, _: &mut MatcherIO) -> bool {
+        let path = file_info.path();
+        let path_str = path.to_string_lossy();
+
+        // This is a quirk in find's traditional semantics probably due to
+        // POSIX rmdir() not accepting "." (EINVAL). std::fs::remove_dir()
+        // inherits the same behavior, so no reason to buck tradition.
+        if path_str == "." {
+            return false;
```

Comment:
`return true` to match other find implementations

Reply:
Nice catch, thanks! It didn't occur to me to chain another matcher afterward to check that... too used to -delete being the end of the line, I guess :)